### PR TITLE
feat!: ERC1155AutoGraphMinter updates to better support Unity 

### PR DESCRIPTION
**Pull request by ghost**
## Description

The Unity client is having issues with calling `ERC1155AutoGraphMinter.sol:mintWithPaymentTokenAsFee` with a tuple. This PR adds a new function that will take the parameters directly, instead of as a tuple.

The contract constructor has a new parameter too, the ERC20 payment token address. We've moved this here as otherwise including it in the function signature of `mintWithPaymentTokenAsFeeWithParams` will result in a `stack too deep` error.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

-   [X] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [X] added `!` to the type prefix if API or client breaking change
-   [ ] ~provided a link to the relevant issue or specification~
-   [ ] included the necessary unit and integration tests
-   [ ] code is adequately commented
-   [ ] updated the relevant documentation or specification
-   [X] reviewed "Files changed" and left comments if necessary
-   [X] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

-   [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [ ] confirmed `!` in the type prefix if API or client breaking change
-   [ ] confirmed all author checklist items have been addressed
-   [ ] reviewed service design and naming
-   [ ] reviewed documentation is accurate
-   [ ] reviewed tests and test coverage
-   [ ] manually tested (if applicable)

### CODE REVIEW

File: src/nfts/ERC1155AutoGraphMinter.sol [hunk 1]
```diff
@@ -45,6 +45,9 @@ contract ERC1155AutoGraphMinter is WhitelistedAddresses, CoreRef, RateLimited {
     /// @notice - expiryToken value for x amount hours
     uint8 public expiryTokenHoursValid; // 1 - 24 hours
 
+    /// @notice payment token
+    address public paymentToken;
```
Comment (utx0) [inline src/nfts/ERC1155AutoGraphMinter.sol]: should we make this `immutable`?

**Review by utx0 — APPROVED** [review]

File: src/nfts/ERC1155AutoGraphMinter.sol (hunk 1 above)
Comment (ghost) [reply src/nfts/ERC1155AutoGraphMinter.sol]: > should we make this `immutable`?

Yes, good spotting, I think we should.

**Review by KBryan — APPROVED** [review]
Looks good.

**Comment by ghost** [comment]
After speaking with @KBryan we determined that this update is no longer required.

**Pull request closed without merging**